src/wallet/create_new_wallet_flow.c

```diff
@@ -326,8 +326,18 @@ new_wallet_state_e new_wallet_state_handler(new_wallet_state_e current_state) {
                         wallet.total_number_of_shares,
                         wallet.minimum_number_of_shares,
                         wallet_shamir_data.mnemonic_shares);
-      if (WALLET_IS_PIN_SET(wallet.wallet_info))
+
+      uint8_t wallet_nonce[NONCE_SIZE] = {0};
+      random_generate(wallet_nonce, 12);
```

Comment:
define a macro for a nonce size of 12

suggestion: could be NONCE_SIZE = 12 and PADDED_NONCE_SIZE = 16

Reply:
updated